core/src/jvmMain/kotlin/io/kinference/compiler/generation/operators/math/MulGenerator.kt

```diff
@@ -0,0 +1,22 @@
+package io.kinference.compiler.generation.operators.math
+
+import io.kinference.compiler.generation.operators.SimpleOperatorGenerator
+import io.kinference.ndarray.arrays.NumberNDArray
+import io.kinference.operators.math.Mul
+
+class MulGenerator(
```

Comment:
You have almost the same classes for + and *. Probably - and / will be almost the same. Can you reuse the common part of the code? For example by using a new abstract class for base math operators?